lib/adapter.js

```diff
@@ -1,5835 +1,5842 @@
-                    if (obj.native && obj.native.devices && oldObj.native && oldObj.native.devices) {
-                        oldObj.native.devices = [];
-                    }
-                    obj = extend(true, oldObj, obj);
-
-                    if (!obj.from) obj.from = 'system.adapter.' + this.namespace;
-                    if (!obj.user) obj.user = (options ? options.user : '') || 'system.user.admin';
-                    if (!obj.ts) obj.ts = Date.now();
-
-                    this.objects.setObject(id, obj, options, callback);
-                });
-            } else {
-                if (!obj.from) obj.from = 'system.adapter.' + this.namespace;
-                if (!obj.user) obj.user = (options ? options.user : '') || 'system.user.admin';
-                if (!obj.ts) obj.ts = Date.now();
-                this.objects.extendObject(id, obj, options, callback);
-            }
-        };
-        /**
-         * Promise-version of Adapter.extendObject
-         */
-        this.extendObjectAsync = tools.promisify(this.extendObject, this);
-
-        /**
-         * Same as {@link Adapter.setObject}, but for any object.
-         *
-         * ID must be specified as a full name with adapter namespace. E.g "hm-rpc.0.ABC98989.1.STATE"
-         *
-         * @alias setForeignObject
-         * @memberof Adapter
-         * @param {string} id object ID, that must be overwritten or created.
-         * @param {object} obj new object
-         * @param {object} options optional user context
-         * @param {function} callback return result
-         *        <pre><code>
-         *            function (err, obj) {
-         *              // obj is {id: id}
-         *              if (err) adapter.log.error('Cannot write object: ' + err);
-         *            }
-         *        </code></pre>
-         */
-        this.setForeignObject = (id, obj, options, callback) => {
-            if (typeof options === 'function') {
-                callback = options;
-                options = null;
-            }
-            if (!obj) {
-                logger.error(`setForeignObject: try to set null object for ${id}`);
-                return callback && callback(tools.ERRORS.ERROR_EMPTY_OBJECT);
-            }
-            if (!obj.from) obj.from = 'system.adapter.' + this.namespace;
-            if (!obj.user) obj.user = (options ? options.user : '') || 'system.user.admin';
-            if (!obj.ts) obj.ts = Date.now();
-            const mId = id.replace(FORBIDDEN_CHARS, '_');
-            if (mId !== id) {
-                logger.warn(`Used invalid characters: ${id} changed to ${mId}`);
-                id = mId;
-            }
-
-            setObjectWithDefaultValue(id, obj, options, callback);
-        };
-        /**
-         * Promise-version of Adapter.setForeignObject
-         */
-        this.setForeignObjectAsync = tools.promisify(this.setForeignObject, this);
-
-        /**
-         * Same as {@link 
```

Comment:
I think this would be easier to read with a separate const for the calculated value.

Reply:
you are right